File: app/src/main/java/com/soonyong/mywebcrawler/config/manage/InMemoryConfigManager.java
package com.soonyong.mywebcrawler.config.manage;

import com.soonyong.mywebcrawler.config.CrawlConfig;

import java.io.IOException;
import java.util.List;

public class InMemoryConfigManager implements ConfigManager {

    private CrawlConfig crawlConfig;

    public InMemoryConfigManager() {
        List<CrawlConfig.Target> targets = new java.util.ArrayList<>();
        targets.add(CrawlConfig.Target.builder()
                .active(true)
                .title("default")
                .interval(1000000)
                .url("http://www.google.com")
                .targetXPath("html")
                .build());
        targets.add(CrawlConfig.Target.builder()
                .active(true)
                .title("default2")
                .interval(1000000)
                .url("http://www.google.com")
                .targetXPath("html")
                .build());
        this.crawlConfig = new CrawlConfig(targets);
    }

    @Override
    public CrawlConfig getCrawlConfig() throws IOException {
        return this.crawlConfig;
    }

    @Override
    public void setCrawlConfig(CrawlConfig crawlConfig) throws IOException {
        this.crawlConfig = crawlConfig;
    }

    @Override
    public void addCrawlConfigTarget(CrawlConfig.Target target) throws IOException {
        this.crawlConfig.getTargets().add(target);
    }
}

File: app/src/main/java/com/soonyong/mywebcrawler/config/manage/InMemoryConfigManager.kt
package com.soonyong.mywebcrawler.config.manage

import com.soonyong.mywebcrawler.config.CrawlConfig
import java.io.IOException
import java.net.URI

class InMemoryConfigManager : ConfigManager {
    override val crawlConfig: CrawlConfig

    init {
        val targets: MutableList<CrawlConfig.Target> = ArrayList()
        targets.add(CrawlConfig.Target(
                active =true,
                title ="ppomppu",
                interval =1000000,
                url= URI("https://www.ppomppu.co.kr/zboard/zboard.php?id=ppomppu"),
                targetXPath="//*[@id=\"revolution_main_table\"]/tbody/tr/td[3]/table/tbody/tr/td[2]/div/a/font"
                ))
        targets.add(CrawlConfig.Target(
                active=true,
                title="default2",
                interval=1000000,
                url=URI("http://www.google.com"),
                targetXPath="html"
                ))
        crawlConfig = CrawlConfig(targets)
    }

    @Throws(IOException::class)
    override fun addCrawlConfigTarget(target: CrawlConfig.Target) {
        crawlConfig.targets.add(target)
    }
}